### PR TITLE
PHP84: ignore deprecated E_STRICT

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

https://php.watch/versions/8.4/E_STRICT-deprecated#:~:text=The%20E_STRICT%20constant%20is%20deprecated,be%20removed%20in%20PHP%209.0

### Related Pull Requests

1. See OpenMage/magento-lts#4286

### CODE REVIEW

**Review by ADDISON74 — APPROVED** [review]

File: app/code/core/Mage/Core/functions.php [hunk 1]
```diff
@@ -165,7 +165,7 @@ function mageCoreErrorHandler($errno, $errstr, $errfile, $errline)
         case E_USER_NOTICE:
             $errorMessage .= 'User Notice';
             break;
-        case E_STRICT:
+        case PHP_VERSION_ID < 80400 && E_STRICT:
```
Comment (kiatng) [inline app/code/core/Mage/Core/functions.php:168]: The `case` is based on `switch($errno)`, which is not a bool comparison.

Comment (sreichel) [reply app/code/core/Mage/Core/functions.php:168]: I dont get it. We switch to operation based on $errno, with an additional condition. I dont have a better idea right now.

Comment (ADDISON74) [reply app/code/core/Mage/Core/functions.php:168]: https://github.com/sebastianbergmann/phpunit/issues/5956

Comment (kiatng) [inline app/code/core/Mage/Core/functions.php:168]:
```suggestion
        case 2048: // E_STRICT prior to PHP8.4
```

Comment (ADDISON74) [reply app/code/core/Mage/Core/functions.php:168]: It is much better using the error value in this case.

**Review by ADDISON74 — DISMISSED** [review]

**Review by ADDISON74 — APPROVED** [review]

**Review by kiatng — APPROVED** [review]

**Pull request merged**